uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -530,28 +578,33 @@ define([
 
       timeObject.loading = true;
       timeObject.disabled = true;
-
-      if (!timeObject.skipValueSetting) {
-        timeObject.time = '';
-        timeObject.amount = '0';
-      }
+      timeObject.time = '';
+      timeObject.amount = '0';
 
       return vm.request.getWorkDayForDate(date)
         .then(function (response) {
           timeObject.min = response.time_from || '00:00';
           timeObject.max = response.time_to || '00:00';
           timeObject.maxAmount = response.number_of_hours.toString() || '0';
+          timeObject.time = (type === 'to' ? timeObject.max : timeObject.min);
+          timeObject.amount = timeObject.maxAmount;
           timeObject.disabled = false;
-
-          if (!timeObject.skipValueSetting) {
-            timeObject.time = (type === 'to' ? timeObject.max : timeObject.min);
-            timeObject.amount = timeObject.maxAmount;
-          }
         })
         .catch(handleError)
         .finally(function () {
           timeObject.loading = false;
-          timeObject.skipValueSetting = false;
+        });
+    }
+
+    /**
+     * Initializes user's calendar (work patterns)
+     *
+     * @return {Promise}
+     */
+    function loadCalendar () {
+      return Calendar.get(vm.request.contact_id, vm.period.start_date, vm.period.end_date)
+        .then(function (usersCalendar) {
+          vm.calendar = usersCalendar;
```

Comment:
shouldn't it be `userCalendar`. Anyway I think in CiviHR we generally refer to "contacts" instead of "users"

Reply:
Done.